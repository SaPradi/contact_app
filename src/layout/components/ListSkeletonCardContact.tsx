import SkeletonCardContact from "./SkeletonCardContact"

const ListSkeletonCardContact:React.FC = () => {
  return (
    <div className="list-contact">
      {
        [...Array(4)].map((_,i)=>(
            <SkeletonCardContact key={i}/>
        ))
      }
    </div>
  )
}

export default ListSkeletonCardContact
