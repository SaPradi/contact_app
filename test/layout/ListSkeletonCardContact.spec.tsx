import React from 'react';
import ListSkeletonCardContact from '../../src/layout/components/ListSkeletonCardContact';
import { render } from '@testing-library/react';

describe('Testing in <ListSkeletonCardContact/>', () => {
  

    test('Should match with the snapshot',()=>{
        const { container } = render(<ListSkeletonCardContact />);
        expect( container ).toMatchSnapshot();
    });

    test('Should show 4 skeleton card elements',()=>{    
        const { getAllByTestId } = render(<ListSkeletonCardContact/>)
        const elements = getAllByTestId('card-skeleton')
        expect( elements.length ).toBe(4) 
    });

})
