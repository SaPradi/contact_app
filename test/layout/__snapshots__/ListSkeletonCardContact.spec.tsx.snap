// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <ListSkeletonCardContact/> Should match with the snapshot 1`] = `
<div>
  <div
    class="list-contact"
  >
    <div
      class="card__skeleton"
      data-testid="card-skeleton"
    >
      <div
        class="card__skeleton__avatar"
      />
      <div
        class="card__skeleton__content"
      >
        <div
          class="card__skeleton__name"
        />
        <div
          class="card__skeleton__email"
        />
      </div>
      <div
        class="card__skeleton__footer"
      >
        <hr />
        <div
          class="card__skeleton__footer__button"
        />
      </div>
    </div>
    <div
      class="card__skeleton"
      data-testid="card-skeleton"
    >
      <div
        class="card__skeleton__avatar"
      />
      <div
        class="card__skeleton__content"
      >
        <div
          class="card__skeleton__name"
        />
        <div
          class="card__skeleton__email"
        />
      </div>
      <div
        class="card__skeleton__footer"
      >
        <hr />
        <div
          class="card__skeleton__footer__button"
        />
      </div>
    </div>
    <div
      class="card__skeleton"
      data-testid="card-skeleton"
    >
      <div
        class="card__skeleton__avatar"
      />
      <div
        class="card__skeleton__content"
      >
        <div
          class="card__skeleton__name"
        />
        <div
          class="card__skeleton__email"
        />
      </div>
      <div
        class="card__skeleton__footer"
      >
        <hr />
        <div
          class="card__skeleton__footer__button"
        />
      </div>
    </div>
    <div
      class="card__skeleton"
      data-testid="card-skeleton"
    >
      <div
        class="card__skeleton__avatar"
      />
      <div
        class="card__skeleton__content"
      >
        <div
          class="card__skeleton__name"
        />
        <div
          class="card__skeleton__email"
        />
      </div>
      <div
        class="card__skeleton__footer"
      >
        <hr />
        <div
          class="card__skeleton__footer__button"
        />
      </div>
    </div>
  </div>
</div>
`;
